synapse/rest/client/v2_alpha/auth.py

```diff
@@ -178,6 +221,40 @@ def on_POST(self, request, stagetype):
             request.write(html_bytes)
             finish_request(request)
 
+            defer.returnValue(None)
+        elif stagetype == LoginType.TERMS:
+            if ('session' not in request.args or
+                    len(request.args['session'])) == 0:
+                raise SynapseError(400, "No session supplied")
+
+            session = request.args['session'][0]
+            authdict = {'session': session}
+
+            success = yield self.auth_handler.add_oob_auth(
+                LoginType.TERMS,
+                authdict,
+                self.hs.get_ip_from_request(request)
+            )
+
+            if success:
+                html = SUCCESS_TEMPLATE
+            else:
+                html = TERMS_TEMPLATE % {
+                    'session': session,
+                    'terms_url': "%s/_matrix/consent" % (
```

Comment:
I'm slightly concerned that a user might end up seeing an old (cached) version of the policy, but that we then record that they have seen the most recent. Could you include the `v=` param in the URI (and then make sure we track the version through the session or sth?)

Reply:
In order for someone to change the version they'd have to restart synapse, which results in the session being lost (there's technically a bug for this somewhere - GitHub on mobile is crap)

I can add the tracking, but it doesn't seem all that useful with the current state of things.

Reply:
sure, but cloudflare or something could cache the consent form unless there is a query param to stop that happening.